src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -403,9 +404,39 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
             return "";
         }
 
-        boolean isContainsNonEmptyResponse = false; // we will only show stats if there is at least one nonempty response
-
         Map<String, Integer> answerFrequency = new LinkedHashMap<String, Integer>();
+        int numChoicesSelected = getQuestionResultStatistics(responses, answerFrequency);
+        if (numChoicesSelected == -1) {
+            return "";
+        } else {
```

Comment:
no its needed. because of this reason :  method returns -1 only in this case when there is no empty responses in the remaining cases it returns the number of nonEmptyResponses stat. Earlier all theses were in the same method but since its extracted out to a common method i am using `if` to check that condition.